Get some real sanity into the linking situation.
- LDLIBS is for libraries only.
- Flags to the linking are passed as LDFLAGS, not LFLAGS.
- LDFLAGS was missed for the 'date' target.

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar Makefile Makefile
--- Makefile	2008-10-14 02:31:04.388508402 -0700
+++ Makefile	2008-10-14 02:36:18.931855910 -0700
@@ -83,7 +83,7 @@
 
 # Non-default libraries needed to link.
 # Add -lintl if you want to use `gettext' on Solaris.
-LDLIBS=$(LDFLAGS)
+LDLIBS=
 
 # Add the following to the end of the "CFLAGS=" line as needed.
 #  -DHAVE_ADJTIME=0 if `adjtime' does not exist (SVR0?)
@@ -309,10 +309,10 @@
 		cp date.1 $(MANDIR)/man1/.
 
 zdump:		$(TZDOBJS)
-		$(CC) $(CFLAGS) $(LFLAGS) $(TZDOBJS) $(LDLIBS) -o $@
+		$(CC) $(CFLAGS) $(LDFLAGS) $(TZDOBJS) $(LDLIBS) -o $@
 
 zic:		$(TZCOBJS) yearistype
-		$(CC) $(CFLAGS) $(LFLAGS) $(TZCOBJS) $(LDLIBS) -o $@
+		$(CC) $(CFLAGS) $(LDFLAGS) $(TZCOBJS) $(LDLIBS) -o $@
 
 yearistype:	yearistype.sh
 		cp yearistype.sh yearistype
@@ -351,7 +351,7 @@
 			then ranlib $@ ; fi
 
 date:		$(DATEOBJS)
-		$(CC) $(CFLAGS) date.o localtime.o asctime.o strftime.o \
+		$(CC) $(CFLAGS) $(LDFLAGS) date.o localtime.o asctime.o strftime.o \
 			$(LDLIBS) -lc -o $@
 
 tzselect:	tzselect.ksh
